features/steps/ruby_notifier_steps.rb

```diff
@@ -1,31 +1,79 @@
-When("I set environment variable {string} to the current IP") do |env_var|
+Then(/^the "(.+)" of the top non-bugsnag stackframe equals (\d+|".+")$/) do |element, value|
+  stacktrace = read_key_path(Server.current_request[:body], 'events.0.exceptions.0.stacktrace')
+  frame_index = stacktrace.find_index { |frame| ! /.*lib\/bugsnag.*\.rb/.match(frame["file"]) }
   steps %Q{
-    When I set environment variable "#{env_var}" to "#{current_ip}"
+    the "#{element}" of stack frame #{frame_index} equals #{value}
   }
 end
 
-When("I set environment variable {string} to the mock API port") do |env_var|
+Then(/^the total sessionStarted count equals (\d+)$/) do |value|
+  session_counts = read_key_path(Server.current_request[:body], "sessionCounts")
+  total_count = session_counts.inject(0) { |count, session| count += session["sessionsStarted"] }
+  assert_equal(value, total_count)
+end
+
+
+### THIS NEEDS TO BE REMOVED BEFORE MERGING
+Then("the request is valid for the error reporting API version {string} for the {string}") do |payload_version, notifier_name|
   steps %Q{
-    When I set environment variable "#{env_var}" to "#{MOCK_API_PORT}"
+    Then the "Bugsnag-Api-Key" header equals "#{$api_key}"
+    And the payload field "apiKey" equals "#{$api_key}"
+    And the "Bugsnag-Payload-Version" header equals "#{payload_version}"
+    And the "Content-Type" header equals "application/json"
+    And the "Bugsnag-Sent-At" header is a timestamp
+
+    And the payload field "notifier.name" equals "#{notifier_name}"
+    And the payload field "notifier.url" is not null
+    And the payload field "notifier.version" is not null
+    And the payload field "events" is a non-empty array
+
+    And each element in payload field "events" has "severity"
+    And each element in payload field "events" has "severityReason.type"
+    And each element in payload field "events" has "unhandled"
+    And each element in payload field "events" has "exceptions"
```

Comment:
Remove?

Reply:
So this step is a response to the [maze-runner version of the step](https://github.com/bugsnag/maze-runner/blob/v2/lib/features/steps/error_reporting_steps.rb#L8) which includes the check `the payload contains the payloadVersion "#{payload_version}"` which will always fail in the Ruby notifier as it does not include the payload version in the payload, only the header.

Upon discussion it was unclear whether this check was necessary for the running of the notifier, and whether or not it could be removed, but no clear action was determined.